components/cache/adapters/memcached_adapter.rst

```diff
@@ -0,0 +1,290 @@
+.. index::
+    single: Cache Pool
+    single: Memcached Cache
+
+Memcached Cache Adapter
+=======================
+
+.. versionadded:: 3.3
+
+    The Memcached adapter was introduced in Symfony 3.3.
+
+This adapter stores the values in-memory using one (or more) `Memcached server`_
+instances. Unlike the ACPu adapter, and similarly to the Redis adapter, it is
+not limited to the current server's shared memory; you can store contents
+independent of your PHP environment. The ability to utilize a cluster of servers
+to provide redundancy and/or fail-over is also available.
+
+.. caution::
+
+    The `Memcached PHP extension`_ as well as a `Memcached server`_ must be
+    installed, active, and running to use this adapter.
+
+This adapter expects a `Memcached`_ instance to be passed as the first
+parameter. A namespace and default cache lifetime can optionally be passed as
+the second and third parameters::
+
+    use Symfony\Component\Cache\Adapter\MemcachedAdapter;
+
+    $cache = new MemcachedAdapter(
+        // the client object that sets options and adds the server instance(s)
+        \Memcached $client,
+
+        // a string prefixed to the keys of the items stored in this cache
+        $namespace = '',
+
+        // the default lifetime (in seconds) for cache items that do not define their
+        // own lifetime, with a value 0 causing items to be stored indefinitely (i.e.
+        // until MemcachedAdapter::clear() is invoked or the server(s) are restarted)
+        $defaultLifetime = 0
+    );
+
+Configure the Connection
+------------------------
+
+The :method:`Symfony\\Component\\Cache\\Adapter\\MemcachedAdapter::createConnection`
+helper method allows creating and configuring a `Memcached`_ class instance using a
+`Data Source Name (DSN)`_ or an array of DSNs::
+
+    use Symfony\Component\Cache\Adapter\MemcachedAdapter;
+
+    // pass a single DSN string to register a single server with the client
+    $client = MemcachedAdapter::createConnection(
+        'memcached://localhost'
+    );
+
+    // pass an array of DSN strings to register multiple servers with the client
+    $client = MemcachedAdapter::createConnection(array(
+        'memcached://10.0.0.100',
+        'memcached://10.0.0.101',
+        'memcached://10.0.0.102',
+        // etc...
+    ));
+
+The `Data Source Name (DSN)`_ for this adapter must use the following format:
+
+.. code-block:: text
+
+    memcached://[user:pass@][ip|host|socket[:port]][?weight=int]
+
+The DSN must include a IP/host (and an optional port) or a socket path, an
+optional username and password (for SASL authentication; it requires that the
+memcached extension was compiled with ``--enable-memcached-sasl``) and an
+optional weight (for prioritizing servers in a cluster; its value is an integer
+between ``0`` and ``100`` which defaults to ``XX``; a higher value means more
```

Comment:
So, there isn't really a "default" for this value. If a weight isn't set, all servers are treated exactly the same. We don't provide any default weight when one isn't specified. How do you think we should make that clear and word it?